test_scripts/Defects/8_1/3838_3_SDL_does_not_send_UI.ClosePopUp_in_case_two_PerformInteraction_requests.lua

```diff
@@ -0,0 +1,142 @@
+local function getPromptValue(pText)
+  return {
+    {
+      text = pText,
+      type = "TEXT"
+    }
+  }
+end
+
+local initialPromptValue = getPromptValue(" Make your choice ")
+
+local helpPromptValue = getPromptValue(" Help Prompt ")
+
+local timeoutPromptValue = getPromptValue(" Time out ")
+
+--[[ Local Functions ]]
+local function setChoiceSet(pChoiceIDValue)
+  local temp = {
+    {
+      choiceID = pChoiceIDValue,
+      menuName ="Choice" .. tostring(pChoiceIDValue),
+      vrCommands = {
+        "VrChoice" .. tostring(pChoiceIDValue),
+      }
+    }
+  }
+  return temp
+end
+
+local function createInteractionChoiceSet(pChoiceSetID)
+  local choiceID = pChoiceSetID
+  local cid = common.getMobileSession():SendRPC("CreateInteractionChoiceSet", {
+      interactionChoiceSetID = pChoiceSetID,
+      choiceSet = setChoiceSet(choiceID),
+    })
+  common.getHMIConnection():ExpectRequest("VR.AddCommand", {
+      cmdID = choiceID,
+      type = "Choice",
+      vrCommands = { "VrChoice" .. tostring(choiceID) }
+    })
+  :Do(function(_, data)
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+    end)
+  common.getMobileSession():ExpectResponse(cid, { resultCode = "SUCCESS", success = true })
+end
+
+local function sendPI(pChoiceVR, pExpResult)
+  local params1 = {
+    initialText = "StartPerformInteraction",
+    initialPrompt = initialPromptValue,
+    interactionMode = "VR_ONLY",
+    interactionChoiceSetIDList = { 100 },
+    helpPrompt = helpPromptValue,
+    timeoutPrompt = timeoutPromptValue,
+    timeout = 5000,
+    vrHelp = { { text = pChoiceVR, position = 1 } }
+  }
+  local cid = common.getMobileSession():SendRPC("PerformInteraction", params1)
+  local success = true
+  if pExpResult ~= "SUCCESS" then success = false end
+  common.getMobileSession():ExpectResponse(cid, { success = success, resultCode = pExpResult })
+end
+
+local function PI_ViaVR_ONLY()
+
+  common.run.runAfter(function() sendPI("ChoiceVR1", "SUCCESS") end, 500)
+  common.run.runAfter(function() sendPI("ChoiceVR2", "REJECTED") end, 1000)
+  common.getHMIConnection():ExpectRequest("UI.ClosePopUp", { methodName = "UI.PerformInteraction" })
+  :Times(0)
+  common.getHMIConnection():ExpectRequest("VR.PerformInteraction")
+  :Do(function(exp, data)
+      if exp.occurences == 1 then
+        common.run.runAfter(function()
+          common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS",
+            { choiceID = 100 })
+          end, 1500)
+      else
+        common.getHMIConnection():SendError(data.id, data.method, "REJECTED",
+          "VR.PerformInteraction is already in progress")
+      end
+    end)
+  :Times(2)
+
+  common.getHMIConnection():ExpectRequest("UI.PerformInteraction")
+  :Do(function(exp, data)
+      if exp.occurences == 1 then
+      common.run.runAfter(function()
+        common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+        end, 1000)
```

Comment:
Doesn't HMI need to response to 1st PI request later than App sends 2nd PI request?
Now 2nd request is sent by App after 1s and exact at the same time HMI responded to 1st request

Reply:
Yes, we have a delay between request/response of 500ms due to the runafter(500) for the first request .. 500+1000 
Anyway, times have been updated for more readable code